rosbag2_compression/src/rosbag2_compression/zstd_decompressor.cpp

```diff
@@ -156,6 +158,52 @@ void write_output_buffer(
   fclose(file_pointer);
 }
 
+/**
+ * Throws a runtime_error if data does not contain a ZSTD compressed frame.
+ */
+void throw_on_invalid_magic_number(const std::vector<uint8_t> & data)
+{
+  if (data.size() < 4) {
+    throw std::runtime_error{"Data is too small to contain a ZSTD frame!"};
```

Comment:
Style and clarity:  "failed to process compressed data: data frame bas size %d which is smaller than %d, the minimum frame size"

^ replace %d by the appropriate specifier